docs/rules/no-mutation-props.md

```diff
@@ -0,0 +1,19 @@
+# Prevent mutation of this.props (no-mutation-props)
+
+NEVER mutate `this.props`, as all React components must act like pure functions with respect to their props. 
```

Comment:
I would be fine with making it stronger, but there's no such thing as "super caps"